velox/functions/sparksql/tests/AtLeastNNonNullsTest.cpp

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/functions/sparksql/specialforms/AtLeastNNonNulls.h"
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+using namespace facebook::velox::test;
+
+static constexpr auto kNaNDouble = std::numeric_limits<double>::quiet_NaN();
```

Comment:
These repeat a lot across tests. Would be nice to introduce some helper class to hold these.

```
FloatConstants::kNanFloat
FloatConstants::kNanDouble
...
```

Reply:
I add the helper class, but there are too many tests use these values, I think it may be better in anohter PR to replace them all. Thanks.